presto-main/src/main/java/com/facebook/presto/metadata/TableLayoutResult.java

```diff
@@ -56,4 +60,28 @@ public boolean hasAllOutputs(TableScanNode node)
 
         return columns.containsAll(nodeColumnHandles);
     }
+
+    public static TupleDomain<ColumnHandle> computeEnforced(TupleDomain<ColumnHandle> tupleDomain, TupleDomain<ColumnHandle> unenforcedTupleDomain)
+    {
+        if (tupleDomain.isNone()) {
+            checkArgument(unenforcedTupleDomain.isNone() || unenforcedTupleDomain.isAll());
+            return TupleDomain.none();
+        }
+        if (unenforcedTupleDomain.isNone()) {
+            return TupleDomain.none();
+        }
+        Map<ColumnHandle, Domain> domains = tupleDomain.getDomains().get();
+        Map<ColumnHandle, Domain> unenforcedDomains = unenforcedTupleDomain.getDomains().get();
+        ImmutableMap.Builder<ColumnHandle, Domain> enforcedDomainsBuilder = ImmutableMap.builder();
+        for (Map.Entry<ColumnHandle, Domain> entry : domains.entrySet()) {
+            ColumnHandle columnHandle = entry.getKey();
```

Comment:
In general, I prefer to assign `entry.getKey` and `entry.getValue` to variables so that they can have a name. But I also acknowledge that in Presto, we prefer strongly toward inlining, as a result, I only do this if a variable is used multiple times, as is the case here.